test/Performance/BuildPerformanceTest.cs

```diff
@@ -0,0 +1,285 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Linq;
+using Microsoft.DotNet.Tools.Test.Utilities;
+using Microsoft.Xunit.Performance;
+using Microsoft.DotNet.TestFramework;
+using System.IO;
+using System.Runtime.CompilerServices;
+using System;
+
+namespace Microsoft.DotNet.Tools.Builder.Tests
+{
+    public class BuildPerformanceTest : TestBase
+    {
+        private static string SingleTargetApp = "SingleTargetApp";
+        private static string TwoTargetApp = "TwoTargetApp";
+
+        private static string[] SingleTargetGraph = new[]
+        {
+            "SingleTargetGraph/SingleTargetP0",
+            "SingleTargetGraph/SingleTargetP1",
+            "SingleTargetGraph/SingleTargetP2"
+        };
+
+        private static string[] TwoTargetGraph = new[]
+        {
+            "TwoTargetGraph/TwoTargetP0",
+            "TwoTargetGraph/TwoTargetP1",
+            "TwoTargetGraph/TwoTargetP2"
+        };
+
+        private static string[] TwoTargetGraphLarge = new[]
+        {
+            "TwoTargetGraphLarge/TwoTargetLargeP0",
+            "TwoTargetGraphLarge/TwoTargetLargeP1",
+            "TwoTargetGraphLarge/TwoTargetLargeP2",
+            "TwoTargetGraphLarge/TwoTargetLargeP3",
+            "TwoTargetGraphLarge/TwoTargetLargeP4",
+            "TwoTargetGraphLarge/TwoTargetLargeP5",
+            "TwoTargetGraphLarge/TwoTargetLargeP6"
+        };
+
+        [Benchmark]
+        public void BuildSingleProject_SingleTargetApp() => BuildSingleProject(CreateTestInstance(SingleTargetApp));
+        [Benchmark]
+        public void BuildSingleProject_TwoTargetApp() => BuildSingleProject(CreateTestInstance(TwoTargetApp));
+
+        public void BuildSingleProject(TestInstance instance)
+        {
+            foreach (var iteration in Benchmark.Iterations)
+            {
+                using (iteration.StartMeasurement())
+                {
+                    new BuildCommand(instance.TestRoot, buildProfile: false)
```

Comment:
Overhead is minimal, we are measuring in seconds, object allocation is quick event compared to process start overhead.